elfi/wrapper.py

```diff
@@ -26,28 +27,31 @@ def __init__(self, command_template="", post=None, pre=None):
 
     @staticmethod
     def process_elfi_internals(command_template, args, kwargs):
-        """ Replace 'prng' in kwargs with a seed from the generator if present in template """
+        """ Replace 'random_state' in kwargs with a seed from the generator if present in template """
         proc_args = list()
         for a in args:
             if isinstance(a, np.ndarray):
-                if a.shape == (1,):
+                if a.shape == (1, 1):
                     # take single values out of array
-                    proc_args.append(a[0])
+                    proc_args.append(a.item())
                 else:
                     raise NotImplementedError("Wrapper does not yet support array arguments")
```

Comment:
I think would be important to have this in. Look for `elfi.tools.vectorize` as how to implement it. Would be great to use same code here and there via some smart way.